meeting.js

```diff
@@ -0,0 +1,346 @@
+        console.log('Left the meeting');
+        cleanUpMediaListeners(currentMeeting);
+        document.getElementById('Video').innerHTML='<img src="./images/stop.png" style="width: 25px; ;height:25px">Start'
+      }).catch((error) => {
+        console.error('Failed to leave the meeting', error);
+      });
+    } else {
+      console.log('User chose not to leave the meeting');
+      document.getElementById('remoteShare').style.display="none";
+    }
+  }
+});
+
+
+// Adding Event Listeners to the JOIN, START/STOP, UNMUTE/MUTE buttons
+document.addEventListener("DOMContentLoaded", () => {
+  document.getElementById("joinMeetingButton").addEventListener("click", joinMeeting);
+  document.getElementById("Video").addEventListener("click", toggleVideo);
+  document.getElementById("Microphone").addEventListener("click", toggleMicrophone);
+  document.getElementById("Microphone").innerHTML = '<img src="./images/mute.png" alt="Mute Icon"> Unmute';
+});
+
+
+
+// JOIN MEETING FUNCTIONALITY
+async function joinMeeting() {
+  document.getElementById("joinMeetingButton").style.display = "none";
+  $('#joinMeetingModal').modal('show');
+  var element = document.querySelector('.container3');
+  element.classList.add('blur');
+  try {
+    const meeting = await webex.meetings.create(sip);
+    currentMeeting = meeting;
+    setMediaListeners(meeting);
+
+    // Load the local video stream before joining the meeting
+    await loadCamera({ video: true, audio: false });
+
+    await meeting.join();
+    
+    // Initialize the video state based on the sendVideo parameter
+    isVideoStarted = true;
+   // const isSharing = meeting.getShareStatus();
+
+    // Add local video (and audio, if needed) to the meeting
+    await meeting.addMedia({
+      localStream: localMedia.cameraStream,
+      mediaSettings: {
+        receiveVideo:true,
+        receiveAudio: true,
+        receiveShare: false,
+        sendVideo: isVideoStarted,
+        sendAudio: false,
+        sendShare: false
+      }
+    });
+    // Update the video button text based on the video state
+    document.getElementById('Video').innerHTML = isVideoStarted ? '<img src="./images/video-camera.png" style="width: 25px; ;height:25px">Stop' : '<img src="./images/stop.png" style="width: 25px;height:25px">Start';
+    document.getElementById("Microphone").innerHTML = '<img src="./images/mute.png" alt="Mute Icon"> Unmute';
+    // UI updates for successfully joining the meeting
+  } catch (error) {
+    console.error('Failed to join or create the meeting', error);
+    alert('Failed to join or create the meeting: ' + error.message);
+  }
+}
+
+// LOAD CAMERA FUNCTIONALITY
+async function loadCamera(constraints) {
+  try {
+    // Stop any existing camera stream
+    if (localMedia.cameraStream) {
+      localMedia.cameraStream.getTracks().forEach(track => track.stop());
+    }
+
+    // Attempt to get a new camera stream
+    const stream = await navigator.mediaDevices.getUserMedia(constraints);

```

Comment:
We should instead be using `createVideoStream` and `createAudioStream` methods from `mediaHelpers`

Please check the [webex-js-sdk-meeting-demo webex.js:138](https://github.com/WebexSamples/webex-js-sdk-meeting-demo/blob/main/webex.js#L138) for reference.